DhsTranslator/src/DhsService.cpp

```diff
@@ -0,0 +1,489 @@
+/*
+ * DhsService.cpp
+ *
+ *  Created on: Feb 12, 2015
+ *      Author: jluhrs
+ */
+
+#include "defines.h"
+#include "DhsService.h"
+#include <axiom_element.h>
+#include <axis2_conf.h>
+#include <sstream>
+#include "DhsAdapter.h"
+#include "DhsAdapterSim.h"
+
+using namespace std;
+using namespace boost;
+
+const char * const DhsService::SERVICE_NAME = "dhs";
+const char * const DhsService::SIM_PARAM = "dhsSimulate";
+const char * const DhsService::MY_NAME_PARAM = "myName";
+const char * const DhsService::DHS_HOST_PARAM = "dhsHost";
+const char * const DhsService::DHS_NAME_PARAM = "dhsName";
+const char * const DhsService::RESPONSE_NODE_NAME = "response";
+const char * const DhsService::STATUS_NODE_NAME = "status";
+const char * const DhsService::RESULT_NODE_NAME = "result";
+const char * const DhsService::ERRORS_NODE_NAME = "errors";
+const char * const DhsService::ERROR_STATUS = "error";
+const char * const DhsService::SUCCESS_STATUS = "success";
+const char * const DhsService::XSI_NAMESPACE_PREFIX = "xsi";
+const char * const DhsService::XSI_NAMESPACE_URI =
+        "http://www.w3.org/2001/XMLSchema-instance";
+const char * const DhsService::ENC_NAMESPACE_PREFIX = "enc";
+const char * const DhsService::ENC_NAMESPACE_URI =
+        "http://schemas.xmlsoap.org/soap/encoding/";
+const char * const DhsService::DHS_ERROR = "DHS_ERROR";
+const char * const DhsService::BAD_REQUEST = "BAD_REQUEST";
+const char * const DhsService::INTERNAL_ERROR = "INTERNAL_SERVER_ERROR";
+
+DhsService *DhsService::theService = NULL;
+const axis2_svc_skeleton_ops_t DhsService::skeletonOps = { DhsService::stubInit,
```

Comment:
Yes. The entry point for the library is `axis2_get_instance()`, but it returns a pointer to that `axis2_svc_skeleton_ops_t` structure.